code/game/dna/dna_modifier.dm

```diff
@@ -512,402 +503,335 @@
+							databuf.dna.real_name = connected.occupant.dna.real_name
+						databuf.name = "Structural Enzymes"
+						buffers[bufferId] = databuf
+				if("clear")
+					buffers[bufferId] = new /datum/dna2/record()
+				if("changeLabel")
+					tgui_modal_input(src, "changeBufferLabel", "Please enter the new buffer label:", null, list("id" = bufferId), buffer.name, TGUI_MODAL_INPUT_MAX_LENGTH_NAME)
+				if("transfer")
+					if(!connected.occupant || (NOCLONE in connected.occupant.mutations && connected.scan_level < 3) || !connected.occupant.dna)
+						return
+
+					irradiating = 2
+					var/lock_state = connected.locked
+					connected.locked = TRUE //lock it
+
+					SStgui.update_uis(src)
+					sleep(2 SECONDS)
+
+					irradiating = 0
+					connected.locked = lock_state
+
+					var/radiation = (rand(20,50) / connected.damage_coeff)
+					connected.occupant.apply_effect(radiation, IRRADIATE, 0)
+
+					if(connected.radiation_check())
+						return
+
+					var/datum/dna2/record/buf = buffers[bufferId]
+
+					if((buf.types & DNA2_BUF_UI))
+						if((buf.types & DNA2_BUF_UE))
+							connected.occupant.real_name = buf.dna.real_name
+							connected.occupant.name = buf.dna.real_name
+						connected.occupant.UpdateAppearance(buf.dna.UI.Copy())
+					else if(buf.types & DNA2_BUF_SE)
+						connected.occupant.dna.SE = buf.dna.SE.Copy()
+						connected.occupant.dna.UpdateSE()
+						domutcheck(connected.occupant, connected)
+				if("createInjector")
+					if(!injector_ready)
+						return
+					if(text2num(params["block"]) > 0)
+						var/list/choices = all_dna_blocks((buffer.types & DNA2_BUF_SE) ? buffer.dna.SE : buffer.dna.UI)
+						tgui_modal_choice(src, "createInjectorBlock", "Please select the block to create an injector from:", null, list("id" = bufferId), null, choices)
+					else
+						create_injector(bufferId, TRUE)
+				if("loadDisk")
+					if(isnull(disk) || disk.read_only)
+						return
+					buffers[bufferId] = disk.buf.copy()
+				if("saveDisk")
+					if(isnull(disk) || disk.read_only)
+						return
+					var/datum/dna2/record/buf = buffers[bufferId]
+					disk.buf = buf.copy()
+					disk.name = "data disk - '[buf.dna.real_name]'"
+		if("wipeDisk")
+			if(isnull(disk) || disk.read_only)
+				return
+			disk.buf = null
+		if("ejectDisk")
+			if(!disk)
+				return
+			disk.forceMove(get_turf(src))
+			disk = null
 
-			disk.buf = buf.copy()
-			disk.name = "data disk - '[buf.dna.real_name]'"
-			//temphtml = "Data saved."
-			return TRUE
+/**
+  * Creates a blank injector with the name of the buffer at the given buffer_id
+  *
+  * Arguments:
+  * * buffer_id - The ID of the buffer
+  * * copy_buffer - Whether the injector should copy the buffer contents
+  */
+/obj/machinery/computer/scan_consolenew/proc/create_injector(buffer_id, copy_buffer = FALSE)
+	set waitfor = FALSE
+	if(buffer_id < 1 || buffer_id > length(buffers))
+		return
 
+	// Cooldown
+	injector_ready = FALSE
+	spawn(30 SECONDS)
+		injector_ready = TRUE
```

Comment:
imo this should be a proper callback but ill leave it up to debates with the maints

Reply:
I'll change it anyway because I don't think we should be using `spawn` or `sleep`